src/DeploymentTemplate.ts

```diff
@@ -141,121 +132,83 @@ export class DeploymentTemplate {
         });
     }
 
-    /**
-     * Get the document text as a string.
-     */
-    public get documentText(): string {
-        return this._documentText;
-    }
-
-    /**
-     * The unique identifier for this deployment template. Usually this will be a URI to the document.
-     */
-    public get documentId(): string {
-        return this._documentId;
-    }
-
-    public get schemaUri(): string | undefined {
-        const schema = this.schemaValue;
-        return schema ? schema.unquotedValue : undefined;
-    }
-
-    public get schemaValue(): Json.StringValue | undefined {
-        return this._schema.getOrCacheValue(() => {
-            const value: Json.ObjectValue | undefined = Json.asObjectValue(this._jsonParseResult.value);
-            if (value) {
-                const schema: Json.StringValue | undefined = Json.asStringValue(value.getPropertyValue("$schema"));
-                if (schema) {
-                    return schema;
-                }
-            }
-
-            return undefined;
-        });
-    }
+    public async getErrors(_associatedParameters: DeploymentParameters | undefined): Promise<language.Issue[]> {
```

Comment:
>getErrors [](start = 17, length = 9)

Not caching this anymore?

Reply:
Added comment to consider caching again, but need to better understand whether the associated doc is part of the permanent state of the current deployment template or not.